pkg/api/types/field_type.go

```diff
@@ -5,21 +5,21 @@ package types
 // cannot have both FieldType.Text and FieldType.Reference defined. Only one is allowed.
 type FieldType struct {
 	// Text represents the configuration for a text field
-	Text *FieldTypeText `json:"text,omitempty"`
+	Text *FieldTypeText `json:"text,omitempty" yaml:"text,omitempty"`
```

Comment:
Same point about `omitempty` in this file

Reply:
This is the same scenario where only 1 of the fields must be non-nil, all the others must be nil... If we remove `omitempty` the api payload will be like
```
fieldType:
  text: {}
  multilineText: null
  date: null
  ...: null
```
I think we prefer
```
fieldType:
  text: {}
```

Reply:
Ah ok. In that case yes it makes sense to omit empty, but I would have structured it slightly differently. Unsure how this would look in go, but in typescript:
```
type FieldType = FieldTypeText | FieldTypeReference | FieldTypeSubForm | FieldTypeMultilineText ....
```
I'm not sure what implications this would have though and isn't a change I'd suggest for this pr.

Reply:
It's what I usually see on Go for implementing polymorphism, otherwise we would need a discriminator like you suggest, and then we would need custom `MarshalJSON` and `UnmarshalJSON` methods on the `FieldType`. imho this is very standard from working with Kubernetes api a lot, where you get things like
```
- name: my-env-var
  value: VALUE
- name: my-config-var
  valueFrom:
    configMapRef: 
      name: my-config-map
      key: config-key
- name: my-config-var
  valueFrom:
    secretKeyRef: 
      name: my-secret
      key: secret-key
- name: my-field-path-var
  valueFrom:
    fieldRef:
      fieldPath: metadata.namespace
    
```

Reply:
For now it's a bit weird because we have a lot of empty objects that don't have any config
```
- fieldType:
    text: {}
- fieldType
    multilineText: {}
```
 but we can imagine (in the future)
```
- fieldType:
    text:
      minLength: 10
      maxLength: 20
- fieldType
    multilineText:
      columns: 5
      maxLength: 3000
```

Reply:
I reviewed the omitempties, removed it for mandatory fields

Reply:
My issue with this type of polymorphism is that this is valid:
```
- fieldType:
    text:
      minLength: 10
      maxLength: 20
    multilineText:
      columns: 5
      maxLength: 3000
```
And also determining the field type on the client side is more difficult than just doing `fieldType.type`.

Reply:
look at https://github.com/nrc-no/core/pull/259/files#diff-4788377ddf8ac7c4bd0ae52920cd70b077b6802c71471d4466b670848ac1b900

Reply:
and https://github.com/nrc-no/core/blob/main/web/app/client/src/lib/utils/getFieldKind.ts

Reply:
> My issue with this type of polymorphism is that this is valid:

It is valid _json_, but its not a valid `FormDefinition`

Reply:
So yeah you can raise errors in the validation, but I prefer having it in the type checking as it's less error prone. But maybe this a better to talk about outside of the pr at some point.